packages-web/signature/src/components/Signature.tsx

```diff
@@ -0,0 +1,116 @@
+import { createElement, PureComponent, ReactNode } from "react";
+
+// @ts-ignore
+import SignaturePad, { IOptions } from "signature_pad";
+import classNames from "classnames";
+import ReactResizeDetector from "react-resize-detector";
+
+import { Alert } from "./Alert";
+import { Grid } from "./Grid";
+import { Dimensions, SizeContainer } from "./SizeContainer";
+
+import "../ui/Signature.scss";
+
+export interface SignatureProps extends Dimensions {
+    className: string;
+    alertMessage?: string;
+    clearSignature: boolean;
+    showGrid: boolean;
+    gridCellWidth: number;
+    gridCellHeight: number;
+    gridBorderColor: string;
+    gridBorderWidth: number;
+    penType: penOptions;
+    penColor: string;
+    onSignEndAction?: (imageUrl?: string) => void;
+    wrapperStyle?: object;
+    readOnly: boolean;
+}
+
+export type penOptions = "fountain" | "ballpoint" | "marker";
+
+export class Signature extends PureComponent<SignatureProps> {
+    // @ts-ignore
```

Comment:
why? should be initial assigned?

Reply:
Tests are failing with: `TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/components/Signature.tsx:33:13 - error TS2564: Property 'canvasNode' has no initializer and is not definitely assigned in the constructor.`